src/views/signup.jsx

```diff
@@ -1,7 +1,185 @@
-import React from "react";
+import { useState } from "react";
+import { Link, useNavigate } from "react-router-dom";
+import {
+  createUserWithEmailAndPassword,
+  GoogleAuthProvider,
+  signInWithPopup,
+} from "firebase/auth";
+import { auth } from "../../firebase_config";
 
-const signup = () => {
-  return <div>signup</div>;
-};
+function Signup() {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [confirmPassword, setConfirmPassword] = useState("");
 
-export default signup;
+  const provider = new GoogleAuthProvider();
+  const navigate = useNavigate();
+
+  const handleSignUp = async (e) => {
```

Comment:
we can safely remove async here since await keyword is not used in the function

Reply:
maybe add some navigation after Email Signup. For example, adding something like `navigate("/dashboard")` after a success signup, similar to what you did for Google sign-up.